### PR TITLE
Jamie/generic graph options

### PR DESCRIPTION
**Pull request by charlesknightsbridge**
Implements generic graph options for colours and axis min/max
DOES NOT implement generic graph option dialog (that's next)

### CODE REVIEW

**Review by charlesknightsbridge — COMMENTED** [review]
Missing unit files were the stat_dialogs for epicurve, histogram and barchart.
I also sorted the units again. Notice that in the next commit, the <OtherUnitFiles> change again with the replacement of graph_utils with chart_options

File: epidataanalysis.lpr [hunk 1]
```diff
@@ -12,7 +12,7 @@
   Interfaces, sysutils, Forms,
 
   // Required EpiData units
-  ana_procs, wizard_form, datamodule, main,
+  ana_procs, wizard_form, datamodule, main, chart_options,
```
Comment (torstenchr) [inline epidataanalysis.lpr:15]: chart_options should automatically be included in other units and is technically not needed here. You are free to decide if you want to leave the unit in or not.

File: src/graph/options/chart_options.pas [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  function ToColors(AValue: String; out Msg: UTF8String): TColorMap;
+  function ToColors(AHexValues: Array of String; out Msg: UTF8String): TColorMap;
+  function ChartColorsFromOptions(OptionList: TOptionList; Executor: TExecutor): TColorMap;
+  function AxisTypeFromVariableType(AVar: TVariableList; AIndex: Integer): TASTResultTypes;
+implementation
+
+uses
+  LazUTF8, ana_globals, typinfo, math, strutils, options_utils;
+
+function ChartColorsFromOptions(OptionList: TOptionList; Executor: TExecutor): TColorMap;
```
Comment (torstenchr) [inline src/graph/options/chart_options.pas:27]: I am not a fan of having a dependency to the TExecutor in this unit when not strictly needed. It creates a coupling in the code that can be hard to remove if we need a refactor later on.

**Review by torstenchr — CHANGES_REQUESTED** [review]
PLease consider the suggested changes. Especially the dependency of TExecutor

File: epidataanalysis.lpr (hunk 1 above)
Comment (charlesknightsbridge) [reply epidataanalysis.lpr]: removed

File: src/graph/options/chart_options.pas (hunk 2 above)
Comment (charlesknightsbridge) [reply src/graph/options/chart_options.pas]: It was a bad strategy to gain access to error messages. So removed. Errors in colours are communicated to calling routine via msg string

**Comment by charlesknightsbridge** [comment]
refactored all graphs to get colours; this is now one of the first actions of a graph execution; any error in colour is recognized by non-empty msg; with error message and exit.

Rats - I meant for the epi_osxlocale change to come in the next PR, but here it is. See PR in core.

**Pull request merged**